core/src/main/java/org/apache/calcite/sql2rel/StandardConvertletTable.java

```diff
@@ -684,7 +686,13 @@ protected RexNode convertCast(
     final boolean safe = kind == SqlKind.SAFE_CAST;
     final SqlNode left = call.operand(0);
     final SqlNode right = call.operand(1);
+    final SqlNode format = call.getOperandList().size() > 2 ?
+        call.operand(2) : null;
+
     final RexBuilder rexBuilder = cx.getRexBuilder();
+    final RexNode arg = cx.convertExpression(left);
+    final RexNode formatArg = isNull(format) ? null : cx.convertExpression(format);
```

Comment:
Why use the `isNull` method, here and elsewhere? `isNull` detects the `NULL` literal, but I don't think that `CAST(d AS DATE FORMAT NULL)` is even allowed (correct me if I'm wrong).

The only null is a java null, where you pass a RexLiteral whose value is null if there was no `FORMAT` clause.

I am wondering whether you intend to support an expression for the format clause, e.g. `CAST(d AS DATE FORMAT 'yyyy-' || 'mm')` but as there are no tests I can't answer that question.

Reply:
Thanks for checking this Julian, I intended to use a Null-valued RexLiteral in cases where the format clause is not specified with the CAST expression. I've updated the flow to populate/check the constant null literal where required.